ucc-demo.ipynb

```diff
@@ -0,0 +1,319 @@
+    "\n",
+    "3. When you first execute the notebook, you will first be prompted to select  `Python Environments...` and then the recommended environment from the `Select Another Kernel` window.\n",
+    "\n",
+    "### Local \n",
+    "If you are running this notebook locally, be sure you have have the `ucc` packaged installed via `pip install ucc` or have followed the [documentation](https://ucc.readthedocs.io/en/latest/#installation) on setting up via poetry."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# imports used throughout this demo\n",
+    "import ucc\n",
+    "from ucc import compile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Quick start\n",
+    "\n",
+    "By leveraging [qBraid](https://github.com/qBraid/qBraid), UCC interfaces automatically with multiple quantum computing frameworks, including [Qiskit](https://github.com/Qiskit/qiskit), [Cirq](https://github.com/quantumlib/Cirq), and [PyTKET](https://github.com/CQCL/tket) and supports programs in OpenQASM 2 and [OpenQASM 3](https://openqasm.com/).\n",
+    "\n",
+    "The list of supported formats is"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ucc.supported_circuit_formats"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's take `ucc` compile for a spin with `pytket` ([docs](https://docs.quantinuum.com/tket/user-guide/index.html)) and `cirq` ([docs](https://quantumai.google/cirq)) on a simple 2-qubit circuit.\n",
+    "\n",
+    "Starting with `pytket`, let's draw the pre and post compiled-circuits:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Using pytket, show the pre and post compiled basic circuit\n",
+    "import pytket\n",
+    "from pytket.circuit.display import render_circuit_jupyter as Tketdraw\n",
+    "\n",
+    "\n",
+    "tket_circuit = pytket.Circuit(2)\n",
+    "tket_circuit.H(0)\n",
+    "tket_circuit.CX(0, 1)\n",
+    "tket_compiled_circuit = compile(tket_circuit)\n",
+    "\n",
+    "Tketdraw(tket_circuit)\n",
+    "Tketdraw(tket_compiled_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's do the same using the `cirq` library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "\n",
+    "qubits = cirq.LineQubit.range(2)\n",
+    "cirq_circuit = cirq.Circuit(\n",
+    "    cirq.H(qubits[0]),\n",
+    "    cirq.CNOT(qubits[0], qubits[1])\n",
+    ")\n",
+    "cirq_compiled_circuit = compile(cirq_circuit)\n",
+    "\n",
+    "print(cirq_circuit)\n",
+    "print(\"------------------------\")\n",
+    "print(cirq_compiled_circuit)"
+   ]
+  },
```

Comment:
This cell resulted in this error for me:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 8
      3 qubits = cirq.LineQubit.range(2)
      4 cirq_circuit = cirq.Circuit(
      5     cirq.H(qubits[0]),
      6     cirq.CNOT(qubits[0], qubits[1])
      7 )
----> 8 cirq_compiled_circuit = compile(cirq_circuit)
     10 print(cirq_circuit)
     11 print("------------------------")

TypeError: compile() missing required argument 'filename' (pos 2)
```

Reply:
Yes, this is due to not having this updated code in the version of `ucc` thats on PyPI. I'll take the README approach and have this instructions reference the proper version to install, and then we can revert after the next UCC release.